drivers/virtualbox/virtualbox_test.go

```diff
@@ -1,10 +1,207 @@
+	assert.Equal(t, driver.Boot2DockerImportVM, "test-boot2docker-vm")
+	assert.Equal(t, driver.HostOnlyCIDR, "192.168.99.1/8")
+	assert.Equal(t, driver.HostOnlyNicType, "test-nic-type")
+	assert.Equal(t, driver.HostOnlyPromiscMode, "accept")
+	assert.Equal(t, driver.NoShare, true)
+}
+
+func TestSSHHostname(t *testing.T) {
+	hostname, err := newDriver("default").GetSSHHostname()
+
+	assert.Equal(t, "127.0.0.1", hostname)
+	assert.NoError(t, err)
+}
+
+func TestDefaultSSHUsername(t *testing.T) {
+	username := newDriver("default").GetSSHUsername()
+
+	assert.Equal(t, "docker", username)
+}
+
+type VBoxManagerMock struct {
+	expectedArgs string
+	output       string
+	stdErr       string
+	err          error
+}
+
+func (v *VBoxManagerMock) queryVirtualBox(args ...string) (string, string, error) {
+	if strings.Join(args, " ") == v.expectedArgs {
+		return v.output, v.stdErr, v.err
+	}
+	return "", "", errors.New("Invalid args")
+}
+
+func TestStateRunning(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="running"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Running, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStatePaused(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="paused"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Paused, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateSaved(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="saved"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Saved, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateStopped(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="poweroff"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Stopped, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateAborted(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="aborted"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.Stopped, machineState)
+	assert.NoError(t, err)
+}
+
+func TestStateUnknown(t *testing.T) {
+	driver := newDriver("default")
+	driver.VBoxManager = &VBoxManagerMock{
+		expectedArgs: "showvminfo default --machinereadable",
+		output:       `VMState="whatever"`,
+	}
+
+	machineState, err := driver.GetState()
+
+	assert.Equal(t, state.None, machineState)
+	assert.NoError(t, err) // Really?
```

Comment:
yes ;-) I'll cover the code as is for now until I get more confidence in changing the actual code